packages/runtime-core/__tests__/rendererElement.spec.ts

```diff
@@ -1,9 +1,55 @@
+import { nodeOps, render, h, serialize, Text } from '@vue/runtime-test'
+
 describe('renderer: element', () => {
-  test.todo('with props')
+  test('simple', () => {
+    const root = nodeOps.createElement('div')
+
+    render(h('div'), root)
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('with props', () => {
+    const root = nodeOps.createElement('div')
+
+    render(
+      h('div', {
+        id: 'test'
+      }),
+      root
+    )
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('with direct text children', () => {
+    const root = nodeOps.createElement('div')
+
+    render(h('div', ['I ❤ Vue']), root)
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('with text node children', () => {
+    const root = nodeOps.createElement('div')
+
+    render(h('div', [h(Text, ['I ❤ Vue'])]), root)
+
+    expect(serialize(root)).toMatchSnapshot()
+  })
+
+  test('handle already mounted VNode', () => {
```

Comment:
This test case is meant to test reusing the same vnode when rendering to the *same* existing tree. i.e. this is for when users reusing a constant vnode in manual component render functions.

The behavior would be more like:

``` js
const vnode = h('div')
const tree1 = h('div', [vnode]
const tree2 = h('div', [vnode]

render(tree1, root)
render(tree2, root)
```

Also need to assert that after the render, `tree2.children[0]` should have been replaced by a clone of `vnode`.

Should